s3_upload/utils/io.py

```
@@ -53,6 +54,12 @@ def acquire_lock(lock_file="/var/log/s3_upload/s3_upload.lock") -> int:
             f" {datetime.now().strftime('%H:%M:%S')} from process"
             f" {os.getpid()}".encode(),
         )
+
+        # register the file lock to be released on exit
+        atexit.register(release_lock, lock_fd=lock_fd)
+
+        return lock_fd
```

Ruff (0.8.0)

Consider moving this statement to an `else` block

(TRY300)